schema/omega-services.async.api.yaml

```diff
@@ -1,525 +1,415 @@
+        The content of the body of the message has failed validation, e.g. a
+        parsed JSON message may be missing some required properties, or those
+        properties may have invalid values.
+      traits:
+        - $ref: '#/components/messageTraits/JMSMessage'
+        - $ref: '#/components/messageTraits/JMSResponse'
+        - $ref: '#/components/messageTraits/OMGMessage'
+      payload:
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
+    ProcessingError:
+      messageId: OELIFERR007
+      name: ProcessingError
+      title: ProcessingError
+      description: >-
+        The received message was valid, but the system was unable complete the
+        processing of a message.
+      traits:
+        - $ref: '#/components/messageTraits/JMSMessage'
+        - $ref: '#/components/messageTraits/JMSResponse'
+        - $ref: '#/components/messageTraits/OMGMessage'
       payload:
-        $ref: "#/components/schemas/errorMessage"
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
+  messageTraits:
+    JMSMessage:
+      name: JMS Message
+      description: Headers that are mostly common to all JMS Messages
+      headers:
+        required:
+          - JMSMessageID
+          - JMSTimestamp
+        properties:
+          JMSMessageID:
+            name: JMSMessageID
+            description: >-
+              A unique identifier for the message. We suggest using a short
+              application instance specific prefix followed by a unique
+              identifier. In the extreme case a UUID may be used.
+            type: string
+          JMSTimestamp:
+            name: JMSTimestamp
+            description: >-
+              The time the message was sent. The value of the timestamp is the
+              amount of time, measured in milliseconds, that has elapsed since
+              midnight, January 1, 1970, UTC.
+            type: number
+    JMSResponse:
+      name: JMS Response Message
+      description: JMS Message sent in response to another JMS Message
+      headers:
+        properties:
+          JMSCorrelationID:
+            name: JMSCorrelationID
+            description: >-
+              An identifier that is used to associate this message as a reply to
+              a previous message. The value is the JMSMessageId of the previous
+              message.
+            type: string
       correlationId:
-        description: Default Correlation ID
-        location: $message.header#/correlationId
-
+        description: >-
+          The value is the JMSMessageId of the previous message to which this
+          message is to be correlated.
+        location: $message.header#/JMSCorrelationID
+    OMGMessage:
+      name: Omega Message
+      description: Headers that are common to all Omega Messages
+      headers:
+        required:
+          - OMGApplicationID
+          - OMGMessageTypeID
+          - OMGMessageFormat
```

Comment:
I believe these "headers" are the application specific properties that we will set via (for example): 
```
jmsMessage.setStringProperty("OMGMessageTypeID","OSLISALS001")
```
It would be nice to have an `examples:` field added to the schema for each of these to make it a bit clearer.

Reply:
These could actually be set as the`JMSType` header - I am just weighing up the pros and cons before making that change...

Reply:
Likewise - I think we could move forward here by leaving it as is for now, and then updating it when/if we can get the necessary changes made ti AsyncAPI.